src/openrct2/localisation/date.c

```diff
@@ -67,6 +67,71 @@ void date_reset()
     gCurrentTicks = 0;
 }
 
+void date_set(sint32 year, sint32 month, sint32 day)
+{
+    year = clamp(1, year, 8192);
+    month = clamp(1, month, (int)MONTH_COUNT);
+    day = clamp(1, day, days_in_month[month - 1]);
+    gDateMonthsElapsed = (year - 1) * MONTH_COUNT + month - 1;
+    gDateMonthTicks = 0x10000 / days_in_month[month - 1] * (day - 1) + 4;
+}
+
+sint32 cmdline_for_date(const utf8 **argv, sint32 argc)
+{
+    sint32 year = 0;
+    sint32 month = 0;
+    sint32 day = 0;
+    if (argc < 1 || argc > 3)
+    {
+        return -1;
+    }
+    if (argc == 1)
+    {
+        year = atoi(argv[0]);
+        if (year < 1 || year > 8192) {
```

Comment:
There is quite a bit of overlap in each of these three cases. I think this could be improved upon by moving the common parts, e.g. getting the year and setting the date, outside of the if-blocks.